### PR TITLE
Feature flags for Solana/Terra; clean up EVM_DISABLED confusion

### PR DESCRIPTION
**Pull request by samsondav**
Note: needs a docs PR

### CODE REVIEW

File: core/services/relay/delegate.go [hunk 1]
```diff
@@ -36,19 +36,17 @@ type delegate struct {
...
-	d.addRelayer(types.Solana, solanaRelayer)
-	d.addRelayer(types.Terra, terraRelayer)
 	return d
 }
 
-// addRelayer registers the relayer r, or a disabled placeholder if nil.
-func (d delegate) addRelayer(n types.Network, r types.Relayer) {
+// AddRelayer registers the relayer r, or a disabled placeholder if nil.
+// NOT THREAD SAFE
+func (d delegate) AddRelayer(n types.Network, r types.Relayer) {
```
Comment (connorwstein) [inline core/services/relay/delegate.go:49]: probably want to remove the disabled relayer with this change, it looks not needed anymore

File: core/config/envvar/schema.go [hunk 2]
```diff
@@ -92,6 +93,12 @@ type ConfigSchema struct {
 	FeatureFeedsManager bool `env:"FEATURE_FEEDS_MANAGER" default:"false"` //nodoc
 	FeatureUICSAKeys    bool `env:"FEATURE_UI_CSA_KEYS" default:"false"`   //nodoc
 
+	// General chains/RPC
+	EVMEnabled    bool `env:"EVM_ENABLED" default:"true"`
+	EVMRPCEnabled bool `env:"EVM_RPC_ENABLED" default:"true"`
```
[the remark this answers is not in the document]
Comment (samsondav) [reply core/config/envvar/schema.go:97]: Almost never, it's kept for backwards compatibility with how ETH_DISABLED used to work.

File: core/config/envvar/schema.go [hunk 3]
```diff
@@ -47,7 +48,7 @@ type ConfigSchema struct {
 	// Database Global Lock
 	AdvisoryLockCheckInterval time.Duration `env:"ADVISORY_LOCK_CHECK_INTERVAL" default:"1s"`
 	AdvisoryLockID            int64         `env:"ADVISORY_LOCK_ID" default:"1027321974924625846"`
-	DatabaseLockingMode       string        `env:"DATABASE_LOCKING_MODE" default:"dual"`
+	DatabaseLockingMode       string        `env:"DATABASE_LOCKING_MODE" default:"advisorylock"`
```
[the remark this answers is not in the document]
Comment (samsondav) [reply core/config/envvar/schema.go:50]: This was a bad merge from a release branch to develop, looks like Jordan fixed it here: https://github.com/smartcontractkit/chainlink/pull/5880/files

File: core/config/envvar/schema.go [hunk 4]
```diff
@@ -100,7 +107,6 @@ type ConfigSchema struct {
 	EthereumURL           string `env:"ETH_URL"`
 	// Global
 	DefaultChainID   *big.Int `env:"ETH_CHAIN_ID"`
-	EVMDisabled      bool     `env:"EVM_DISABLED" default:"false"`
```
[the remark this answers is not in the document]
Comment (samsondav) [reply core/config/envvar/schema.go:103]: This is removed and replaced by EVM_ENABLED

File: core/config/envvar/schema.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmank88) [reply core/config/envvar/schema.go:50]: develop is out of sync https://github.com/smartcontractkit/chainlink/pull/5880

**Comment by jmank88** [comment]
It looks like `FEATURE_` is already in use on a few old vars, but we could consider something less verbose like `X_` for experimental.

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably bbd41d94e6f76ff7f46cd22988bb0e8c103141bd) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/config/envvar/schema.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (connorwstein) [reply core/config/envvar/schema.go:97]: What would be the rare use case though? Is there no way to deprecate it and then after a period of time remove it completely

File: core/config/envvar/schema.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (connorwstein) [reply core/config/envvar/schema.go:103]: ok looks like EthereumDisabled is now also removed 👍

File: core/config/envvar/schema.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (samsondav) [reply core/config/envvar/schema.go:97]: I think it's actually used in some tests. It's kind of useful to be able to boot up, add jobs etc but not have the node do any real RPC calls.

File: core/services/relay/delegate.go (hunk 1 above)
Comment (samsondav) [reply core/services/relay/delegate.go:49]: Done

File: core/chains/evm/chain_set.go [hunk 5]
```diff
@@ -24,7 +24,8 @@ import (
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
-var ErrNoChains = errors.New("no chains loaded, are you running with EVM_DISABLED=true?")
+// ErrNoChains indicates that no chains have been started
+var ErrNoChains = errors.New("no chains loaded")
```
Comment (jmank88) [inline core/chains/evm/chain_set.go:28]: Will this always be wrapped/logged with something that specifies `evm`? Or would that be important to include here?

File: core/services/keystore/master.go [hunk 6]
```diff
@@ -25,6 +25,8 @@ import (
 
 var ErrLocked = errors.New("Keystore is locked")
 
+type defaultEVMChainIDFunc = func() (defaultEVMChainID *big.Int, err error)
```
Comment (jmank88) [inline core/services/keystore/master.go:28]: Is this equals necessary? Is this technically declaring an alias to an anonymous type? (I think that's effectively equivalent anyways :shrug:)
```suggestion
type defaultEVMChainIDFunc func() (defaultEVMChainID *big.Int, err error)
```

Comment (samsondav) [reply core/services/keystore/master.go:28]: Did not know you could do that

**Review by jmank88 — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Review by connorwstein — APPROVED** [review]

File: core/config/envvar/schema.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (connorwstein) [reply core/config/envvar/schema.go:97]: For testing, either the geth sim, local geth node or a mock should have us covered?

**Pull request merged**